Bottom: f2bfd59d60ee80b90e33f4d71269d72de85e8735
Top:    ce27d0f314b669323ff98c057197416a215b8042
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-21 02:05:50 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 819d4b5..c02e07d 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -227,6 +227,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
